core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -777,15 +777,17 @@ public static Jenkins getInstanceOrNull() {
      * @return The instance.
      * @throws IllegalStateException {@link Jenkins} has not been started, or was already shut down
      */
+    @SuppressFBWarnings(value="NP_NONNULL_RETURN_VIOLATION", justification="TODO delete kill switch ~2.110")
     @CLIResolver
     @Nonnull
     public static Jenkins getInstance() {
         Jenkins instance = HOLDER.getInstance();
         if (instance == null) {
-            if(SystemProperties.getBoolean(Jenkins.class.getName()+".enableExceptionOnNullInstance")) {
-                // TODO: remove that second block around 2.20 (that is: ~20 versions to battle test it)
-                // See https://github.com/jenkinsci/jenkins/pull/2297#issuecomment-216710150
-                throw new IllegalStateException("Jenkins has not been started, or was already shut down");
+            IllegalStateException ise = new IllegalStateException("Jenkins has not been started, or was already shut down");
+            if (Boolean.getBoolean(Jenkins.class.getName() + ".disableExceptionOnNullInstance")) {
```

Comment:
Inside this condition I would add a second level if that checks Version and will throw the ISE if the version is > 2.110 so that the kill switch is hard disabled on Jenkins 2.110+ (because we forgot to make the changes in 2.20 that we were supposed to)

Reply:
-1 on that. This would cause a behavioral change merely due to `maven-release-plugin`, which would be an unwelcome surprise.

I will just set a calendar reminder for myself. :-)

Reply:
Everybody is welcome to contribute to the API/SystemProperty deprecation engine though

Reply:
FTR superseded by #3195

Reply:
(I did in fact have a calendar entry and it took me a while to remember what it was for…)